src/main/java/org/openrewrite/staticanalysis/TernaryOperatorsShouldNotBeNested.java

```diff
@@ -0,0 +1,88 @@
+package org.openrewrite.staticanalysis;
+
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Recipe;
+import org.openrewrite.Tree;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.JavaVisitor;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.JRightPadded;
+import org.openrewrite.java.tree.Space;
+import org.openrewrite.java.tree.Statement;
+import org.openrewrite.marker.Markers;
+
+
+public class TernaryOperatorsShouldNotBeNested extends Recipe {
+    @Override
+    public String getDisplayName() {
+        return "Ternary operators should not be nested";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Just because you can do something, doesn’t mean you should, and that’s the case with nested ternary operations. " +
+                "Nesting ternary operators results in the kind of code that may seem clear as day when you write it, " +
+                "but six months later will leave maintainers (or worse - future you) scratching their heads and cursing.\n\n" +
+                "Instead, err on the side of clarity, and use another line to express the nested operation as a separate statement.";
+    }
+
+    @Override
+    public Duration getEstimatedEffortPerOccurrence() {
+        return Duration.ofMinutes(5);
+    }
+
+
```

Comment:
Would be nice to override `getTags()` here to show [RSPEC-3358](https://rules.sonarsource.com/java/RSPEC-3358).